test/e2e/kube_state_metrics_test.go

```diff
@@ -65,3 +72,138 @@ func TestKSMMetricsSuppression(t *testing.T) {
 	}
 
 }
+
+func TestKSMCRSMetrics(t *testing.T) {
+	assetsDir := "./assets/"
+	query := "kube_customresource_kube_verticalpodautoscaler_labels_info"
+
+	// Install a VPAv1 CRD.
+	manifest, err := f.ReadManifest(assetsDir + "verticalpodautoscalers-v1-crd.yaml")
+	if err != nil {
+		t.Fatalf("failed to read VPA CRD manifest: %v", err)
+	}
+	vpaCRD, err := f.BuildCRD(manifest)
+	if err != nil {
+		t.Fatalf("failed to build VPA CRD: %v", err)
+	}
+	createVPACRD(t, vpaCRD)
+
+	// Create a VPAv1 CR.
+	vpaCR := &vpav1.VerticalPodAutoscaler{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-vpa",
+			Namespace: f.Ns,
+			Labels: map[string]string{
+				"foo": "bar",
+			},
+		},
+		Spec: vpav1.VerticalPodAutoscalerSpec{
+			TargetRef: &v1.CrossVersionObjectReference{
+				APIVersion: "apps/v1",
+				Kind:       "Deployment",
+				Name:       "cluster-monitoring-operator",
+			},
+		},
+	}
+	createVPACR(t, vpaCR)
+
+	// Fetch KSM CRS metrics.
+	fetchKSMCRSMetrics(t, false, query)
+
+	// Cleanup.
+	deleteVPACR(t, vpaCR)
+	deleteVPACRD(t, vpaCRD)
+
+	// Fetch KSM CRS metrics, but expect failure.
+	fetchKSMCRSMetrics(t, true, query)
+}
+
+func fetchKSMCRSMetrics(t *testing.T, shouldfail bool, query string) {
+	err := framework.Poll(time.Second, time.Minute, func() error {
+		response, err := f.PrometheusK8sClient.PrometheusQueryWithStatus(query, 200)
+		if err != nil {
+			return err
+		}
+		if shouldfail {
+			if strings.Contains(string(response), "foo") {
+				return errors.New(fmt.Sprintf("unexpected label 'foo' found in response: %s", string(response)))
+			}
+		} else {
+			if !strings.Contains(string(response), "foo") {
+				return errors.New(fmt.Sprintf("expected label 'foo' not found in response: %s", string(response)))
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		t.Fatalf("failed to query Prometheus: %v", err)
+	}
+}
+
+func createVPACR(t *testing.T, vpaCR *vpav1.VerticalPodAutoscaler) {
+	_, err := f.VPAClient.VerticalPodAutoscalers(f.Ns).Create(ctx, vpaCR, metav1.CreateOptions{})
```

Comment:
I'd rather Poll on the custom resource creation. We don't expect any error eventually (e.g. "AlreadyExists" isn't an option) and we can exit as soon as the API told us that the creation was successful.